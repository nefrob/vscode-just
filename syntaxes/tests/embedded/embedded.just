# Embedded languages
# TODO: how to test embedded grammars in snapshots?

python:
    #!/usr/bin/env python3
    print('Hello from python!')

js:
    #!/usr/bin/env node
    console.log('Greetings from JavaScript!')

perl:
    #!/usr/bin/env perl
    print "Larry Wall says Hi!\n";

sh:
    #!/usr/bin/env sh
    hello='Yo'
    echo "$hello from a shell script!"

ruby:
    #!/usr/bin/env ruby
    puts "Hello from ruby!"

empty-lines:
    #!/usr/bin/env python3
    print('Hello from python!')

    print('Hello from python!')
