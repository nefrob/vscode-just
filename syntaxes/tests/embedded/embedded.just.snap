># Embedded languages
#^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
># TODO: how to test embedded grammars in snapshots?
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>python:
#^^^^^^ source.just entity.name.function.just
#      ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env python3
#^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>  print('Hello from python!')
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>js:
#^^ source.just entity.name.function.just
#  ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env node
#^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>  console.log('Greetings from JavaScript!')
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>perl:
#^^^^ source.just entity.name.function.just
#    ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env perl
#^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>  print "Larry Wall says Hi!\n";
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>sh:
#^^ source.just entity.name.function.just
#  ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env sh
#^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>  hello='Yo'
#^^^^^^^^^^^^^ source.just meta.recipe.body.just
>  echo "$hello from a shell script!"
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>ruby:
#^^^^ source.just entity.name.function.just
#    ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env ruby
#^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>  puts "Hello from ruby!"
#^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just